x/auth/vesting/types/msgs.go

```diff
@@ -75,3 +80,63 @@ func (msg MsgCreateVestingAccount) GetSigners() []sdk.AccAddress {
 	}
 	return []sdk.AccAddress{from}
 }
+
+// NewMsgCreatePermanentLockedAccount returns a reference to a new MsgCreatePermanentLockedAccount.
+//nolint:interfacer
+func NewMsgCreatePermanentLockedAccount(fromAddr, toAddr sdk.AccAddress, amount sdk.Coins) *MsgCreatePermanentLockedAccount {
+	return &MsgCreatePermanentLockedAccount{
+		FromAddress: fromAddr.String(),
+		ToAddress:   toAddr.String(),
+		Amount:      amount,
+	}
+}
+
+// Route returns the message route for a MsgCreatePermanentLockedAccount.
+func (msg MsgCreatePermanentLockedAccount) Route() string { return RouterKey }
+
+// Type returns the message type for a MsgCreatePermanentLockedAccount.
+func (msg MsgCreatePermanentLockedAccount) Type() string { return TypeMsgCreatePermanentLockedAccount }
+
+// ValidateBasic Implements Msg.
+func (msg MsgCreatePermanentLockedAccount) ValidateBasic() error {
+	from, err := sdk.AccAddressFromBech32(msg.FromAddress)
+	if err != nil {
+		return err
+	}
+	to, err := sdk.AccAddressFromBech32(msg.ToAddress)
+	if err != nil {
+		return err
+	}
+	if err := sdk.VerifyAddressFormat(from); err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid sender address: %s", err)
+	}
+
+	if err := sdk.VerifyAddressFormat(to); err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid recipient address: %s", err)
+	}
+
+	if !msg.Amount.IsValid() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, msg.Amount.String())
+	}
+
+	if !msg.Amount.IsAllPositive() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, msg.Amount.String())
+	}
+
+	return nil
+}
+
+// GetSignBytes returns the bytes all expected signers must sign over for a
+// MsgCreatePermanentLockedAccount.
+func (msg MsgCreatePermanentLockedAccount) GetSignBytes() []byte {
+	return sdk.MustSortJSON(amino.MustMarshalJSON(&msg))
+}
+
+// GetSigners returns the expected signers for a MsgCreatePermanentLockedAccount.
+func (msg MsgCreatePermanentLockedAccount) GetSigners() []sdk.AccAddress {
+	from, err := sdk.AccAddressFromBech32(msg.FromAddress)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
```suggestion
	from, _ := sdk.AccAddressFromBech32(msg.FromAddress)
```

the validate basic method will already ensure we have a valid address, no need to check error